front/post_upsert_hooks/hooks/lib/data_source_helpers.ts

```diff
@@ -0,0 +1,181 @@
+import { CoreAPI } from "@app/lib/core_api";
+import { Diff, diffStrings } from "@app/lib/diff";
+import { DataSource, Workspace } from "@app/lib/models";
+
+export async function getPreviousDocumentVersion({
```

Comment:
To simplify this part could we create an API route on Core that does exactly what we need? 
(sorry if that is a stupid question, I'm new 😬 )

Reply:
I think we could indeed, but I am wondering if it wouldn't be a bit too specific to our use-case
"get version of document right before that specific version hash" may not be useful outside of post upsert hooks. 

Or maybe the "get versions" endpoint could take a specific hash that would act as a "start hash" ? i.e, get all versions older or equal to this hash (and default to the latest if not provided)

Reply:
Going to add to `core` 👍

Reply:
Did it in a separate PR: https://github.com/dust-tt/dust/pull/796

Will update here once merged

Reply:
Overall much simpler now, thank you for the tip !